service/history/replication/stream_sender.go

```diff
@@ -525,7 +527,9 @@ Loop:
 				return nil
 			}
 			task.Priority = priority
-			s.logger.Debug("StreamSender send replication task", tag.TaskID(task.SourceTaskId))
+			if s.isTieredStackEnabled {
+				s.flowController.Wait(priority)
```

Comment:
Do we need to reset context timeout after RESUME?

Reply:
Good catch. The context used here is for `GetReplicationTasksIter` call, and entire pagination is sharing the same context. We may need to revisit the pagination code to reset context there.

Reply:
Discussed offline, the context fix will be in next PR, which will change the signature of GetReplicationTasksIter.